server/util/clickhouse/schema/schema.go

```diff
@@ -179,8 +179,18 @@ type Execution struct {
 	DiskWriteOperations int64
 
 	// Task sizing
-	EstimatedMemoryBytes int64
-	EstimatedMilliCPU    int64
+	EstimatedMemoryBytes   int64
+	EstimatedMilliCPU      int64
+	EstimatedFreeDiskBytes int64
+	RequestedMemoryBytes   int64
+	RequestedMilliCPU      int64
+	RequestedFreeDiskBytes int64
```

Comment:
do we store enough information to reconstruct whether these came from `EstimatedComputeUnits` or from `EstimatedCPU` / `EstimatedMemory`? At some point we probably want to simplify EstimatedCPU/EstimatedMemory so that they are actually respected (instead of only respected when EstimatedComputeUnits is set or if the workload-isolation-type is firecracker), and knowing what those values are set to would help us understand the impact of that change

Reply:
This isn't part of task size, so I got it by calling platform.ParseProperties.

Reply:
What I'm getting at is that I'm not sure if we can use the current DB columns to differentiate between (A) the user set EstimatedComputeUnits, or (B) the user set only EstimatedCPU and EstimatedMemoryBytes? (I think the answer is no but maybe I'm missing something.)

Here is the context for why I think this is interesting data: we have this logic to skip using the measured task size if either (A) `EstimatedComputeUnits` is set: https://github.com/buildbuddy-io/buildbuddy/blob/2348120722abe758bcdaecf0a2016790a149b739/enterprise/server/tasksize/tasksize.go#L154-L157 or if (B) the isolation type is firecracker: https://github.com/buildbuddy-io/buildbuddy/blob/2348120722abe758bcdaecf0a2016790a149b739/enterprise/server/tasksize/tasksize.go#L165-L167

Case (A) is really unfortunate because it's confusing - I think the user expectation would be that if they set EstimatedComputeUnits _or_ EstimatedCPU _or_ EstimatedMemoryBytes then those should take precedence over the measured task size. We even document these as "overrides" in our public docs (https://www.buildbuddy.io/docs/rbe-platforms#runner-resource-allocation), but that documentation actually isn't correct. `EstimatedComputeUnits` are overrides, but `EstimatedCPU` and `EstimatedMemoryBytes` are effectively just "hints" (except in the firecracker case).

So, to see if we can untangle ourselves from this confusing situation, I wonder if we'll have some way to query whether there are any users that are _relying_ on the current behavior. If no users are relying on it, then we can update the actual behavior to match the docs, without breaking anyone.

Reply:
Sorry, I made a change but forgot to push it. Take another look. What I'm doing now is getting the requested sized directly from the action/command/platform, and including a separate field for compute units.